### DISCUSSION TITLE
Passing request based information (headers) to the environment

### DISCUSSION THREAD

**Discussion by roderik** [coder/code-server#4325] [Q&A]
I have a code-server running with some shared pieces of code, e.g. to manage our infrastructure as code setup. While there is no real concurrent use, a few of us use it to update the infrastructure. 

Right now, we interact with services via our JWT based authentication layer, which is also in front of the code server instance. Meaning the request to code server has an Authorization header.

The code inside code-server, also needs that JWT token of the user browsing to it, to interact with other services. 

Not sure how this could work, but it would be awesome if these headers could be transformed in some way to an env variable that the infra code could pick up from the terminal environment. 

The reason I do not want to use a code-server per user, is that this would involve more resources, and making sure that nobody forgets to sync the git repo in code server before they deploy.

Not sure this is even possible, expect if the webserver service code server could modify the global environment, but at the same time I'm unclear on how the terminal inside codeserver actually instantiates/works.

**Comment by jsjoeio** [comment]
This is a really interesting question and use case. Let's see if @bpmct @code-asher or @jawnsy have any thoughts.

**Comment by code-asher** [comment]
I think the closest thing we have is that we store query variables in the config file so they can be used by local processes.

https://github.com/cdr/code-server/pull/1920
